File: 204_singleton/singleton.cpp
/*
@Copyright:LintCode
@Author:   ultimate010
@Problem:  http://www.lintcode.com/problem/singleton
@Language: C++
@Datetime: 16-08-03 15:51
*/

class Solution {
public:
    /**
     * @return: The same instance of this class every time
     */
    static Solution* getInstance() {
        // write your code here
        static Solution * point = NULL;
        if (point == NULL)
            point = new Solution();
        return point;
    }
};

File: 204_singleton/singleton.py
# coding:utf-8
'''
@Copyright:LintCode
@Author:   ultimate010
@Problem:  http://www.lintcode.com/problem/singleton
@Language: Python
@Datetime: 16-06-15 06:25
'''

class Solution:
    # @return: The same instance of this class every time
    curInstance = None
    @classmethod
    def getInstance(cls):
        # write your code here
        if cls.curInstance is None:
            cls.curInstance = Solution()
        return cls.curInstance